Tests/Public/Core/DatabaseQueue/DatabaseQueueTests.swift

```diff
@@ -28,4 +28,154 @@ class DatabaseQueueTests: GRDBTestCase {
+                    XCTFail("Expected Error")
+                }
+                catch let error as DatabaseError {
+                    // expected error
+                    XCTAssertEqual(error.code, 1) // SQLITE_ERROR
+                    XCTAssertEqual(error.message!.lowercaseString, "no such function: succ") // lowercaseString: accept multiple SQLite version
+                    XCTAssertEqual(error.sql!, "SELECT succ(1)")
+                    XCTAssertEqual(error.description.lowercaseString, "sqlite error 1 with statement `select succ(1)`: no such function: succ")
+                }
+            }
+        }
+    }
+    
+    func testAddRemoveCollation() {
+        // Adding a collation and then removing it should succeed
+        assertNoError {
+            do {
+                let dbQueue = try makeDatabaseQueue()
+                let collation = DatabaseCollation("test_collation_foo") { (string1, string2) in
+                    return (string1 as NSString).localizedStandardCompare(string2)
+                }
+                dbQueue.addCollation(collation)
+                try dbQueue.inDatabase { db in
+                    try db.execute("CREATE TABLE files (name TEXT COLLATE TEST_COLLATION_FOO)")
+                }
+                dbQueue.removeCollation(collation)
+                do {
+                    try dbQueue.inDatabase { db in
+                        try db.execute("CREATE TABLE files_fail (name TEXT COLLATE TEST_COLLATION_FOO)")
+                        XCTFail("Expected Error")
+                    }
+                    XCTFail("Expected Error")
+                }
+                catch let error as DatabaseError {
+                    // expected error
+                    XCTAssertEqual(error.code, 1) // SQLITE_ERROR
+                    XCTAssertEqual(error.message!.lowercaseString, "no such collation sequence: test_collation_foo") // lowercaseString: accept multiple SQLite version
+                    XCTAssertEqual(error.sql!, "CREATE TABLE files_fail (name TEXT COLLATE TEST_COLLATION_FOO)")
+                    XCTAssertEqual(error.description.lowercaseString, "sqlite error 1 with statement `create table files_fail (name text collate test_collation_foo)`: no such collation sequence: test_collation_foo")
+                }
+            }
+        }
+    }
+    
+    func testConfigurationSetGet() {
+        assertNoError {
+            do {
+                let inputConfiguration = dbConfiguration
+                let dbQueue = try makeDatabaseQueue()
+                var queueConfiguration = dbQueue.configuration
+                
+                // Test values of the Configurations for equality
+                // NOTE: some of the values can't be tested for equality (ex. closures)
+                
+                XCTAssertEqual(inputConfiguration.foreignKeysEnabled, queueConfiguration.foreignKeysEnabled)
+                XCTAssertEqual(inputConfiguration.readonly, queueConfiguration.readonly)
```

Comment:
Please remove this line. It is fragile.

What needs to be tested are the side-effects of providing a readonly configuration (inability to write), but not the conservation of this flag.

One could imagine that readonly would be set to true if the database file is locked. We already change input configuration sometimes: https://github.com/groue/GRDB.swift/blob/v0.70.1/GRDB/Core/DatabasePool.swift#L64